src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -1609,13 +1609,20 @@ static QVariant fcnRegexpReplace( const QVariantList &values, const QgsExpressio
   QString str = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
   QString regexp = QgsExpressionUtils::getStringValue( values.at( 1 ), parent );
   QString after = QgsExpressionUtils::getStringValue( values.at( 2 ), parent );
+  bool isValid_after{ values.at( 2 ).isValid() };
```

Comment:
```suggestion
  bool isValid_after{ values.at( 2 ).isValid() && !values.at( 2 ).isNull() };
```

Not sure of myself here. But I'm also wondering if just the isNull() condition wouldn't be enough ?

In your test, if you replace QVariant with QVariant( QMetaType::String ) for instance, it would be valid and null, and so your fix wouldn't work, no?

Reply:
We should always use QgsVariantUtils::isNull for these checks.